pkg/eval/platform/platform.go

```diff
@@ -35,9 +38,41 @@ import (
 // Whether or not the platform is Microsoft Windows.
 // This is read-only.
 
+//elvdoc:fn hostname &strip-domain=false
+//
+// The host name of the system. If option `&strip-domain` is used any domain
+// components are stripped from the host name. Domain components include the
+// first dot (period) and anything that follows. For example,
+// `machname.subdomain.tld` => `machname`.
+//
+// If an error occurs while fetching the host name an exception is thrown.
+// This is implemented using Go's
+// [`os.Hostname`](https://golang.org/pkg/os/#Hostname) function.
+
+var osHostname = os.Hostname // to allow mocking in unit tests
+var errCannotDetermineHostname = errors.New("cannot determine the hostname")
+
+type hostnameOpt struct{ StripDomain bool }
+
+func (o *hostnameOpt) SetDefaultOptions() {}
+
+func hostname(opts hostnameOpt) (string, error) {
+	hostname, err := osHostname()
+	if err != nil {
+		return "", errCannotDetermineHostname
```

Comment:
It's better to just preserve the original error.

Reply:
I think it's better to wrap the original error but I didn't know how to do that. If the person using the `platform:hostname` command wants to trap just failures to determine the hostname they shouldn't have to know about all the possible ways the underlying implementation can fail.